src/components/DeeplinkWrapper/index.website.js

```diff
@@ -0,0 +1,155 @@
+import PropTypes from 'prop-types';
+import React, {PureComponent} from 'react';
+import deeplinkRoutes from './deeplinkRoutes';
+import FullScreenLoadingIndicator from '../FullscreenLoadingIndicator';
+import TextLink from '../TextLink';
+import * as Illustrations from '../Icon/Illustrations';
+import LogoWordmark from '../../../assets/images/expensify-wordmark.svg';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import Text from '../Text';
+import styles from '../../styles/styles';
+import CONST from '../../CONST';
+import CONFIG from '../../CONFIG';
+import Icon from '../Icon';
+import colors from '../../styles/colors';
+import * as Browser from '../../libs/Browser';
+
+const propTypes = {
+    /** Children to render. */
+    children: PropTypes.node.isRequired,
+
+    ...withLocalizePropTypes,
+};
+
+class DeeplinkWrapper extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            appInstallationCheckStatus: this.isMacOSWeb() ? 'checking' : 'not-installed',
+        };
+    }
+
+    componentDidMount() {
+        if (this.isMacOSWeb()) {
+            let focused = true;
+
+            window.addEventListener('blur', () => {
+                focused = false;
+            });
+
+            setTimeout(() => {
+                if (!focused) {
+                    this.setState({appInstallationCheckStatus: 'installed'});
+                } else {
+                    this.setState({appInstallationCheckStatus: 'not-installed'});
+                }
+            }, 500);
+
+            // check if pathname matches with deeplink routes
+            const pathname = window.location.pathname;
+            const matchedRoute = _.find(deeplinkRoutes, (route) => {
+                const routeRegex = new RegExp(route.pattern);
+                return routeRegex.test(pathname);
+            });
+
+            if (matchedRoute) {
+                this.setState({deeplinkMatch: true});
+
+                this.openRouteInDesktopApp();
+            } else {
+                this.setState({deeplinkMatch: false});
+            }
+        }
+
+        return undefined;
+    }
+
+    openRouteInDesktopApp() {
+        const pathname = window.location.pathname;
+        const expensifyUrl = new URL(CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL);
+        const expensifyDeeplinkUrl = `${CONST.DEEPLINK_BASE_URL}${expensifyUrl.host}${pathname}`;
+
+        // This check is necessary for Safari, otherwise, if the user
+        // does NOT have the Expensify desktop app installed, it's gonna
+        // show an error in the page saying that the address is invalid
+        if (CONST.BROWSER.SAFARI === Browser.getBrowser()) {
+            const iframe = document.createElement('iframe');
+            iframe.style.display = 'none';
+            document.body.appendChild(iframe);
+            iframe.contentWindow.location.href = expensifyDeeplinkUrl;
+
+            setTimeout(() => {
```

Comment:
@cowboycito Why and we don't prefer using setTimeout? 

cc: @NikkiWines

Reply:
This is so the iframe has enough time to do the deeplink request. This is exactly how Slack and Zoom do on Safari.

Reply:
bump @NikkiWines any thoughts?

Reply:
Sorry, was OOO for the end of the year. It seems fine to me, we do prefer to avoid `setTimeout` but if it's conditional logic needed for a single browser, I think it's ok. Maybe adding a comment for clarity would be good?

Reply:
Bump on this comment ^^

Reply:
Done